app/forms/idv/inherited_proofing/va/form.rb

```diff
@@ -1,163 +1,34 @@
 module Idv
   module InheritedProofing
     module Va
-      class Form
-        include ActiveModel::Model
-
+      class Form < Idv::InheritedProofing::BaseForm
```

Comment:
I'm not sure I understand why we would introduce inheritance here while there is only one subclass? It seems like it introduces indirection and I'm not sure I see the benefit. This class looks a little prettier but there is complexity hiding in the superclass. Now a reader has to reference both places to understand what is going on.

Reply:
I hear you. I needed to split this up per the PR comment above. TLDR: the user PII is (#user_pii) is service provider-*specific* and I didn't want a user to have to refactor what would certainly be a _very heavy_ lift later on, since we are trying to keep our inherited proofing code _somewhat_ service-provider agnostic. The inheritance chain will never go beyond one generation, and the benefit will certainly outweigh the cost in the end IMHO. This is only the first of several changes I need to make. I have the 2nd of 3 changes I am going to push in a little bit, I think it will make more sense.

Reply:
For example:

https://github.com/18F/identity-idp/pull/7012/files#diff-3316a6462ad5cc81ddc835c7d5a86652545ed8cb44e674e7ddcae5c85ca02379

The service provider-specific form returned from #inherited_proofing_form will be #submit_ted_ and the #user_pii obtained from the form used to populate flow_session[:pii_from_user] in one of our flow steps prior to handoff to the IDV workflow.

Reply:
I still don't think I see it. Since we only have 1 service provider providing inherited proofing at the moment we have to make assumptions about how future service providers will work. That makes me think we will likely need to refactor this no matter how we build it. With that in mind I'm suspicious of adding the indirection that comes with trying to generalize it through inheritance.

I can approve this but with the note that I foresee the complexity here being a problem in the future.

I also added a comment to the change you linked about because I don't think I understand the direction there.

Reply:
@jmhooper that's fine for now, thank you. The only assumption I am trying to make with our Form codez is that we _cannot_ make assumptions about how our future service providers, at least by way of what they will return to us; this is why I used a little meta programming to build/return the hash passed from the Service class. 

I'd like to understand more about your concerns about "how future service providers will work"; what exactly are you concerned about in regards to the way we coded our Form class, because all it is doing is accepting a hash from our Service class? If you could take the time to answer at some point, I'd appreciate understanding this more because I don't want to be blind to anything.